enclave/core/sgx/enter.S

```diff
@@ -50,6 +50,24 @@ oe_enter:
     mov %rsp, %gs:td_host_rsp
     mov %rbp, %gs:td_host_rbp
 
+    // Set the MXCSR according to the Linux x86_64 ABI
+    mov $ABI_MXCSR_INIT, %r10
+    push %r10
+    ldmxcsr (%rsp)
+    pop %r10
+
+    // Set the FPU Control Word according to the Linux x86_64 ABI
+    mov $ABI_FPUCW_INIT, %r10
+    push %r10
+    fldcw (%rsp)
+    pop %r10
```

Comment:
Unfortunately, rsp still points to host stack at this time.
We could define a macro to initialize these values and then call the macro in exception_entry, nested_entry and clean_entry paths after the enclave rsp has been setup.

Reply:
Good catch ... instead of putting it in 3 different places, could we just put it into the start of `call_function` instead? That's where all of the entries dovetail after restoring the stack pointer.

Reply:
Nice idea. 
I checked that none of the instruction in different control paths leading to `call_function` can be influenced by host supplied values of these flags/registers, and therefore it is safe change.